### PR TITLE
[bugfix] Convert Syslog handler port parameter into integer

### PR DESCRIPTION
**Pull request by difey**
In our cluster, we use syslog to collect reframe tests data. We have found SysLogHandler require int type port instead of str type in codes now. It works fine in our cluster after change.

### CODE REVIEW

**Comment by difey** [comment]
> Can I test this patch?

yes

**Comment by teojgo** [comment]
ok to test

File: reframe/core/logging.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def _create_syslog_handler(site_config, config_prefix):
     except ValueError:
         pass
     else:
-        address = (host, port)
+        address = (host, int(port))
```
Comment (vkarak) [inline reframe/core/logging.py:225]: We should handle the case when `port` cannot be converted to an `int`.

Comment (victorusu) [reply reframe/core/logging.py:225]: That's true.

**Pull request merged**